### ISSUE TITLE
Ordered taxonomies don't play together with querystring widget.

### ISSUE THREAD

**Issue by tomgross** [collective/collective.taxonomy#50]
The querystring widget does JSON serializing of his catalog queries. With this change https://github.com/collective/collective.taxonomy/commit/cb7f4ab27b6ca6a4c6ae2551e34a53a0f3ee3690#diff-bda08834e1e6892e7ded74b296fff53dR190 this serialization breaks and the querystring widget is not displayed at all.

This is the content of the querystring: 
```
{'operations': [u'plone.app.querystring.operation.selection.is'], 'group': u'Taxonomy', 'description': u'', 'vocabulary': u'collective.taxonomy.test', 'title': u'Test', 'operators': {u'plone.app.querystring.operation.selection.is': {'widget': u'MultipleSelectionWidget', 'operation': u'plone.app.querystring.queryparser._equal', 'description': u'Tipp: SIe k\xf6nnen * zur automatischen Vervollst\xe4ndigung nutzen.', 'title': u'Ist'}}, 'enabled': True, 'values': OrderedDict([(<BTrees.IOBTree.IOBTree object at 0x7f08ef4a03b0>, {'title': ''}), (1, {'title': u'Informatik'}), ('1', {'title': u'Informatik'})]), 'sortable': False}
```

The traceback is 
```
Traceback (innermost last):
  Module ZPublisher.Publish, line 138, in publish
  Module ZPublisher.mapply, line 77, in mapply
  Module ZPublisher.Publish, line 48, in call_object
  Module plone.app.content.browser.query, line 17, in __call__
  Module json, line 244, in dumps
  Module json.encoder, line 207, in encode
  Module json.encoder, line 270, in iterencode
TypeError: keys must be a string
```

**Comment by vincentfretin** [comment]
cc @malthe

**Comment by malthe** [comment]
Let me try and fix.

**Comment by malthe** [comment]
Fixed in 54923bf02c7827be78c66b3bfd3c98510be0c3a7.

cc @vincentfretin, @tomgross

**Comment by tomgross** [comment]
I recocnized it on a DEV box. I don't need a migration step from my side. Thanks!
